alacritty_terminal/src/config/mod.rs

```diff
@@ -418,17 +401,50 @@ impl Default for DefaultTrueBool {
     }
 }
 
+fn defaulted<T, E>(err: E) -> T
+    where T: Default, E: Display
+{
+    error!("Problem with config: {}; using default value", err);
+    T::default()
+}
+
 pub fn failure_default<'a, D, T>(deserializer: D) -> Result<T, D::Error>
 where
     D: Deserializer<'a>,
     T: Deserialize<'a> + Default,
 {
-    let value = serde_yaml::Value::deserialize(deserializer)?;
-    match T::deserialize(value) {
-        Ok(value) => Ok(value),
-        Err(err) => {
-            error!("Problem with config: {}; using default value", err);
-            Ok(T::default())
-        },
-    }
+    Ok(T::deserialize(Value::deserialize(deserializer)?).unwrap_or_else(defaulted))
+}
+
+pub fn none_str_as_none<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
```

Comment:
I think the naming of this deserializer is still a bit unfortunate. The name would suggest to me that it just deserializes the `none` string to `None` and that's it. It doesn't really encode that it parses anything else generically too and None is just an option.

Maybe something like `deserialize_option_explicit_none` might be better?

Reply:
It follows the same format as `failure_default`: `<condition>_<result>`.

Reply:
Failure default wasn't really written with that in mind though. It's just a good way to describe all things going on with this deserializer. In case of failure, go with the default.

None str as none does not do that. It's easy to confuse the name and think it might do other things than it really does. The double-use of 'none' also is confusing to read and it's hard to even understand exactly what is meant without reading it multiple times. Using `str`, which is also a Rust type is additionally confusing, since that does not actually exist for serde/yaml.